pandas/core/arrays/sparse/array.py

```diff
@@ -834,14 +834,24 @@ def _first_fill_value_loc(self):
         if not len(indices) or indices[0] > 0:
             return 0
 
-        diff = indices[1:] - indices[:-1]
-        return np.searchsorted(diff, 2) + 1
+        # a number larger than 1 should be appended to
+        # the last in case of fill value only appears
+        # in the tail of array
+        diff = np.r_[np.diff(indices), 2]
+        return indices[(diff > 1).argmax()] + 1
```

Comment:
We only want to find the first missing/fill value position, there are some redundant steps. Cythonize needed?

Reply:
I think it's ok and better to keep simple without `Cython`. Like you mentioned in the comment you added to `unique`, `indices` should be very small relative to total array length